src/Analysis/Ast/Impl/Analyzer/Evaluation/ExpressionEval.Scopes.cs

```diff
@@ -49,6 +49,12 @@ public void DeclareVariable(string name, IMember value, VariableSource source, N
         public void DeclareVariable(string name, IMember value, VariableSource source, Location location, bool overwrite = true) {
             var member = GetInScope(name);
             if (member != null && !overwrite) {
+                if (source == VariableSource.Import) {
+                    // Duplicate declaration so if variable gets overwritten it can still be retrieved. Consider:
+                    //    from X import A
+                    //    class A(A): ...
+                    CurrentScope.DeclareImported(name, value, location);
+                }
                 return;
             }
             if (source == VariableSource.Import && value is IVariable v) {
```

Comment:
In the conditional below, shouldn't you also declare anything with variable source import as an imported variable to be consistent? Otherwise the collection of imported variables only consists of those that would override already existing variables and not include those that just happened to be declared in other modules.

Reply:
Probably... I guess like in `base.py`
```python
class A: ...
a = A
```
and in `Main.py`
```python
from Base import a
class a(a): ...
```